category.go

```diff
@@ -77,3 +77,48 @@ func (c *commandCategory) VisibleCommands() []*Command {
 	}
 	return ret
 }
+
+// FlagCategories is a slice of *FlagCategory.
+type FlagCategories []*FlagCategory
```

Comment:
map[category]Flag is better ?

Reply:
Agreed!

Reply:
Welp, I started trying to switch the implementation to `map[string][]Flag`, and it seems non-trivial to me so far... but I'm also nervous about introducing `FlagCategories` and `FlagCategory` and being committed to their behaviors for the rest of the `v2` series 😐